### ISSUE TITLE
Links (as part of regular text) not distinguishable enough

### ISSUE THREAD

**Issue by patrickhlauke** [twbs/bootstrap#15304]
For aesthetic reasons, by default Bootstrap links have no underlines, relying instead just on a color difference from surrounding text.

This in itself violates WCAG 2.0 "1.4.1 Use of color" http://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color

There is a technique suggesting that sufficient difference in color contrast/luminance may be enough in certain conditions, but having some additional cue is preferred http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140916/G183

There are also situations where links are even less distinguishable.

![bootlink2](https://cloud.githubusercontent.com/assets/895831/5315172/0aebe90a-7c72-11e4-9dff-bd7b5a7cee72.PNG)

http://getbootstrap.com/components/#alerts-links - link is of course distinguishable from surrounding text, but the difference between the actionable link and the bold text at the start of the alert ("Well done!" etc) is again only down to a subtle variation in color ... this is confusing for users, as it's not immediately clear what is and isn't a link.

![bootlink3](https://cloud.githubusercontent.com/assets/895831/5315175/143cf68e-7c72-11e4-9e3f-c12ea8e5a16c.PNG)

http://getbootstrap.com/components/#navbar-links No distinction at all between inert text and link, unless the user goes on a "let me just hover my mouse over every word" pixel hunt. All sighted users are affected by this (though at least keyboard users will more readily discover these links by virtue of tabbing onto them).

In short: I propose that by default, links should ideally have some subtle additional indication that they're links. Of course, for links in navigation bars etc, this does not apply...the context there is clear enough ("this is a series of actionable things"). But particularly for links that are just part of regular text, we should really do something.

Failing that, this needs to be noted in our accessibility guidance.

/cc @cvrebert

**Comment by xi** [comment]
Somehow this never landed in v4, Can this issue be reopend or should I file a new one?

**Comment by patrickhlauke** [comment]
oh, well spotted @xi ... forgot all about this. will see if i can re-PR for v4

**Comment by mdo** [comment]
See #30389 :).
